src/Fabulous/AttributeDefinitions.fs

```diff
@@ -17,14 +17,15 @@ type IScalarAttributeDefinition =
     abstract member CompareBoxed: a: obj * b: obj -> ScalarAttributeComparison
 
 /// Attribute definition for scalar properties
-type ScalarAttributeDefinition<'inputType, 'modelType> =
+type ScalarAttributeDefinition<'inputType, 'modelType, 'valueType> =
     { Key: AttributeKey
       Name: string
       Convert: 'inputType -> 'modelType
+      ConvertValue: 'modelType -> 'valueType
```

Comment:
One thing I was thinking about if we can make convert functions optional (`ValueOption`). My understanding that it is `id` function mist of the time. If that is true, skipping calling it with checking for `ValueSome` is probably worth it in terms of perf.

BUT! it is probably going to be tricky to type properly, given that then we will have to convert `'modelType -> 'valueType`. 

So maybe a union can be a solution there.
just a sketch:
```FSharp
type ScalarAttributeDefinition<'inputType, 'modelType, 'valueType> =
  | NoConverters of .... // 'inputType = 'modelType = 'valueType, are all the same
  | JustModelConverted of ... //  'inputType <> 'modelType ,  'valueType ='modelType
  | JustValueConverter of ... // 'inputType = 'modelType ,  'valueType <> 'modelType
  | ModelAndValueConverters of ... // 'inputType <> 'modelType <> 'valueType, are all different
```

Feels very clunky but possible worth considering

Reply:
It's an interesting idea.
The question I think is more like is it worth it, compared to all the extra operations of extracting those convert functions each time we need them?

`id` is a simple `fun x -> x` function
Do you know if in terms of perf calling a function is worse than using pattern matching on `ValueOption`?

Reply:
as far as I know it is "dynamic dispatch" call, and if we use `ValueOption` it should be able to check it on the stack before calling. 

So the worst case scenario: It is `id` function called via `dynamic dispatch`. So it is probably be 100-200 slower than making a check.

Note that if we **DO** need to convert we still have to call the function via "dynamic dispatch".

I think we can see it on benchmarks we have. We have no converters in TestUI, thus we can see the impact of "best case scenario" of this optimization. Probably worth doing separately in the future as a part of the dedicated task for that

Reply:
I opened an issue for it, so we can find a proper way to handle this